tests/deployment/test_client_server.py

```diff
@@ -291,12 +374,20 @@ def check_client_server_inference(
     q_y_pred_encrypted_serialized = fhe_model_server.run(q_x_encrypted_serialized, evaluation_keys)
 
     # Client side : Decrypt, de-quantize and post-process the result
-    q_y_pred = fhe_model_client.deserialize_decrypt(q_y_pred_encrypted_serialized)
-    y_pred = fhe_model_client.deserialize_decrypt_dequantize(q_y_pred_encrypted_serialized)
+    q_y_pred = fhe_model_client.deserialize_decrypt(*to_tuple(q_y_pred_encrypted_serialized))
+    y_pred = fhe_model_client.deserialize_decrypt_dequantize(
+        *to_tuple(q_y_pred_encrypted_serialized)
+    )
 
     # Dev side: Predict using the model and circuit from development
     q_x_test = model.quantize_input(x_test)
-    q_y_pred_dev = model.fhe_circuit.encrypt_run_decrypt(q_x_test)
+    if (
+        ciphertext_format == CiphertextFormat.TFHE_RS
+    ):  # fhe_circuit.server._tfhers_specs is not None:
+        q_y_pred_dev = model.encrypt_run_decrypt_tfhers_concrete(q_x_test)
```

Comment:
The api is starting to diverge since we don't have the fhe_circuit anymore. I find this a bit confusing.